### PR TITLE
Add support for Intel GPU to DCGAN Example

### PR DESCRIPTION
**Pull request by dggaytan**
- Enabled XPU usage in main.py file
- Added XPU usage flag to README.md file

### CODE REVIEW

File: dcgan/README.md [hunk 1]
```diff
@@ -20,31 +20,34 @@ python download.py -c bedroom
 ## Usage
 
 ```
-usage: main.py [-h] --dataset DATASET --dataroot DATAROOT [--workers WORKERS]
-               [--batchSize BATCHSIZE] [--imageSize IMAGESIZE] [--nz NZ]
-               [--ngf NGF] [--ndf NDF] [--niter NITER] [--lr LR]
-               [--beta1 BETA1] [--cuda] [--ngpu NGPU] [--netG NETG]
-               [--netD NETD] [--mps]
+usage: main.py [-h] --dataset DATASET [--dataroot DATAROOT] [--workers WORKERS] [--batchSize BATCHSIZE] [--imageSize IMAGESIZE]
```
Comment (dvrogozh) [inline dcgan/README.md]: I think they tried to limit the line width at 80 characters. I think you better do the same otherwise it won't fit into github pages and scroll bar will appear. Just manually break it at 80 chars or copy from the terminal with width limited to 80.

File: dcgan/main.py [hunk 2]
```diff
@@ -107,19 +111,29 @@
 dataloader = torch.utils.data.DataLoader(dataset, batch_size=opt.batchSize,
                                          shuffle=True, num_workers=int(opt.workers))
 use_mps = opt.mps and torch.backends.mps.is_available()
+use_xpu = opt.xpu and torch.xpu.is_available()
+
+##xpu support
```
Comment (dvrogozh) [inline dcgan/main.py]: drop the comment
```suggestion
```

File: dcgan/main.py [hunk 3]
```diff
@@ -107,19 +111,29 @@
...
 if opt.cuda:
     device = torch.device("cuda:0")
 elif use_mps:
     device = torch.device("mps")
+elif use_xpu:
+    device = torch.device("xpu")
+    print("Number of devices: ", torch.xpu.device_count())
```
Comment (dvrogozh) [inline dcgan/main.py]: drop. seems to be a debug print.

File: dcgan/main.py [hunk 4]
```diff
@@ -107,19 +111,29 @@
...
 
+
 ngpu = int(opt.ngpu)
 nz = int(opt.nz)
 ngf = int(opt.ngf)
 ndf = int(opt.ndf)
 
 
+print("DEVICE TO USE: ", device)
+print(ngpu)
```
Comment (dvrogozh) [inline dcgan/main.py]: drop

File: dcgan/main.py [hunk 5]
```diff
@@ -281,3 +299,5 @@ def forward(self, input):
     # do checkpointing
     torch.save(netG.state_dict(), '%s/netG_epoch_%d.pth' % (opt.outf, epoch))
     torch.save(netD.state_dict(), '%s/netD_epoch_%d.pth' % (opt.outf, epoch))
+
```
Comment (dvrogozh) [inline dcgan/main.py]: drop

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: dcgan/main.py (hunk 2 above)
Comment (dggaytan) [reply dcgan/main.py]: done

File: dcgan/main.py (hunk 3 above)
Comment (dggaytan) [reply dcgan/main.py]: it was, thanks
done

File: dcgan/main.py (hunk 4 above)
Comment (dggaytan) [reply dcgan/main.py]: done

File: dcgan/main.py (hunk 5 above)
Comment (dggaytan) [reply dcgan/main.py]: done

File: dcgan/README.md (hunk 1 above)
Comment (dggaytan) [reply dcgan/README.md]: done
